pkg/clustermesh/common/remote_cluster.go

```diff
@@ -132,6 +136,19 @@ func (rc *remoteCluster) releaseOldConnection() {
 }
 
 func (rc *remoteCluster) restartRemoteConnection() {
+	connectionEstablished := false
+	// Applicable to kvstoremesh only:
+	// We execute a goroutine to implement a timeout on connection.
+	// If the timeout has been reached and a connection hasn't been established, we
+	// remove the cluster from the SyncState to prevent it from blocking readiness forever.
+	go func() {
```

Comment:
This will not work as expected if `restartRemoteConnection` is executed again for whatever reason (e.g., triggered by the watchdog), as there would be leaked goroutines potentially decreasing the counter more than desired on timeout. Similarly, it does not account for the possibility that one of the clusters is removed before connection establishment, as it would still wait for timeout expiration.